js/flow/src/telemetry.ts

```diff
@@ -79,13 +79,15 @@ export function writeFlowSuccess(flowName: string, latencyMs: number) {
 
   const paths = traceMetadataAls.getStore()?.paths || new Set<PathMetadata>();
   if (paths) {
-    const relevantPaths = Array.from(paths).filter((meta) =>
-      meta.path.includes(flowName)
+    const relevantPaths = new Map(
+      Array.from(paths)
+        .filter((meta) => meta.path.includes(flowName))
+        .map((path) => [path.path, path])
     );
 
     logger.logStructured(`Paths[/${flowName}]`, {
       flowName: flowName,
-      paths: relevantPaths.map((p) => p.path),
+      paths: [...relevantPaths.keys()],
     });
 
     relevantPaths.forEach((p) => {
```

Comment:
I don't think this takes into account the new data structure

Reply:
Could you say more here, not sure what you mean

Reply:
Ah, I didn't expect that first arg of forEach was the entry value. So I guess this is fine.